### ISSUE TITLE
Move shape function of convolution to shared files

### ISSUE THREAD

**Issue by zhouxin913** [openxla/stablehlo#361]
### Request description

As follow up of https://github.com/openxla/stablehlo/pull/360

**Comment by zhouxin913** [comment]
blocked by https://github.com/openxla/stablehlo/issues/213

**Comment by burmako** [comment]
Duplicate of #270.
